src/box/lua/config/utils/schema.lua

```diff
@@ -1599,6 +1607,83 @@ end
 
 -- }}} <schema object>:pairs()
 
+-- {{{ <schema object>:jsonschema()
+
+-- This function sets common fields such as `description`, `default`, and `enum`
+-- for a given schema object. These fields are typical in JSON schemas.
+local function set_common_jsonschema_fields(res, schema)
+    res.description = schema.description
+    res.default = schema.default
+    res.enum = schema.allowed_values
+    return setmetatable(res, {
+        __serialize = 'map',
+    })
+end
+
+local function jsonschema_impl(schema, ctx)
+    if is_scalar(schema) then
+        -- If the schema is a scalar type (string, number, etc.)
+        -- copy the corresponding JSON schema.
+        local scalar_copy = table.copy(scalars[schema.type].jsonschema)
+        return set_common_jsonschema_fields(scalar_copy, schema)
+    elseif schema.type == 'record' then
+        local properties = {}
+        for field_name, field_def in pairs(schema.fields) do
+            walkthrough_enter(ctx, field_name)
+            if type(field_name) ~= 'string' then
+                walkthrough_error(ctx, 'JSON does not support non-string keys')
+            end
+            properties[field_name] = jsonschema_impl(field_def, ctx)
+            walkthrough_leave(ctx)
+        end
+        return set_common_jsonschema_fields({
+            type = 'object',
+            properties = properties,
+            additionalProperties = false
+        }, schema)
+    elseif schema.type == 'map' then
+        walkthrough_enter(ctx, '*')
+        if schema.key.type ~= 'string' then
+            walkthrough_error(ctx, 'JSON does not support non-string keys')
+        end
+        local res = set_common_jsonschema_fields({
+            type = 'object',
+            additionalProperties = jsonschema_impl(schema.value, ctx)
+        }, schema)
+        walkthrough_leave(ctx)
+        return res
+    elseif schema.type == 'array' then
+        walkthrough_enter(ctx, '*')
+        local res = {
+            type = 'array',
+            items = jsonschema_impl(schema.items, ctx),
+        }
+        walkthrough_leave(ctx)
+        if schema.validate == validate_no_repeat then
+            res.uniqueItems = true
+        end
+        return set_common_jsonschema_fields(res, schema)
+    else
+        assert(false)
+    end
+end
+
+-- Convert a schema to JSON schema format.
+--
+-- This method generates a Lua table that represents the current schema
+-- in the JSON Schema format. It recursively processes the schema
+-- structure and transforms it into a format compliant with the
+-- JSON Schema specification.
+function methods.jsonschema(self)
+    local ctx = walkthrough_start(self, {})
+    local jsonschema = jsonschema_impl(rawget(self, 'schema'), ctx)
+    jsonschema['$schema'] = 'https://json-schema.org/draft/2020-12/schema'
```

Comment:
I don't think it makes much sense to add a global constant here. The schema URL would only change if we decide to incorporate new features introduced in a newer JSON Schema draft. In that case, we would need to update the `<schema>:jsonschema()` implementation itself as well.
So, creating a constant for the URL wouldn't add much value since any update would involve modifying the method's logic, not just the URL.

Reply:
If you insist, I am generally OK.